Jenkinsfile_k8s

```diff
@@ -0,0 +1,62 @@
+def terraformTemplate = """
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: terraform
+    image: hashicorp/terraform:0.13.5
+    command:
+    - sleep
+    args:
+    - infinity
+    volumeMounts:
+      - name: workspace-volume
+        mountPath: /home/jenkins/agent
+    workingDir: "/home/jenkins/agent"
+  volumes:
+      - name: "workspace-volume"
+        persistentVolumeClaim:
+          claimName: "jenkins-worker-pvc"
+          readOnly: false
+"""
+
+pipeline {
+    agent none
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr: '10'))
+        timeout(time: 30, unit: 'MINUTES')
+        timestamps()
+    }
+
+    environment {
+        LANG = 'en_US.UTF-8'
+        LC_CTYPE = 'en_US.UTF-8'
+        /* These environment variables make it feasible for Git to clone properly while
+         * inside the wacky confines of a Docker container
+         */
+        GIT_COMMITTER_EMAIL = 'git@example.com'
+        GIT_COMMITTER_NAME = 'Git'
+        GIT_AUTHOR_NAME = 'Git'
+        GIT_AUTHOR_EMAIL = 'git@example.com'
+    }
+
+    stages {
+      stage('Terraform Validate') {
+        kubernetes {
+          yaml terraformTemplate
+          defaultContainer 'terraform'
```

Comment:
Newbie questions:
- Is the `kubernetes` block can be defined on top-pipeline level to spawn the pod 1 time, and each stage might specify a different container of the pod to run inside?
- If I provide a `Dockerfile` to provide a custom and reproducible build environment (with the exact tooling I need: `terraform`, `tfsec`, etc.), what would be the pattern to use it with k8s agents?